food-waste-express-api/routes/home.js

```diff
@@ -134,15 +138,35 @@ router.post('/:objectId', async (req, res) => {
     
     try {
         let userBasket = await getUserBasket(currentUserId) //returns first instance
+        let tempBasket = []
+
         if (userBasket != null) {
-            userBasket.addUnique("basketOfProductId", {productId, quantity})
-            await userBasket.save()
+            let doesNotHaveDuplicate = true;
+            if (userBasket.get("basketOfProducts").length > 0) {
+                for (productDetail of userBasket.get("basketOfProducts")) {
+                    if (productDetail.productId == productId) {
+                        let newQuantity = productDetail.quantity + quantity
+                        tempBasket.push({ "productId": productId, "quantity": newQuantity.toString() })
+                        doesNotHaveDuplicate = false;
+                    }
+                    else {
+                        tempBasket.push(productDetail)
+                    }
+                }
+                userBasket.removeAll("basketOfProducts", { productId, quantity })
+                userBasket.set("basketOfProducts", tempBasket)
+                await userBasket.save()
+            }
+            if (doesNotHaveDuplicate) {
```

Comment:
What is this doing? The userBasket.addUnique is unclear what the function would do. Also, didn't we handle the duplicates in the basket earlier?